### PR TITLE
Adding the NIN operator and fixing the space before the operators in …

### PR DESCRIPTION
**Pull request by vorjdux**
…the command /j

### CODE REVIEW

File: postgresql/src/postgresql/convert_sql.cpp [hunk 1]
```diff
@@ -1032,9 +1032,8 @@ auto zpt::pgsql::get_query(zpt::json _in, std::string& _queryr) -> void {
 									}
 									_v += zpt::pgsql::escape(_e);
 								}
-								_queryr += zpt::pgsql::escape_name(_key) +
-									   _found->second + std::string("(") + _v +
-									   std::string(")");
+								_queryr += zpt::pgsql::escape_name(_key) + std::string(" ") 
+									+ _found->second + std::string("(") + _v + std::string(")");
```
Comment (naazgull) [inline postgresql/src/postgresql/convert_sql.cpp]: Please, also add a space after the operator.

**Comment by vorjdux** [comment]
@naazgull done. Can you check again? Thank you :)

**Comment by vorjdux** [comment]
@naazgull Thank you! It looks much better following all the same pattern :) Could you please check again?

File: postgresql/src/postgresql/convert_sql.cpp [hunk 2]
```diff
@@ -915,288 +932,412 @@ auto zpt::pgsql::get_query(zpt::json _in, std::string& _queryr) -> void {
...
 
 				if (_part[0] == '-') {
 					_dir = "DESC";
 				}
+
 				_part.erase(0, 1);
-				_queryr += (!_first ? ", " : "") + zpt::pgsql::escape(zpt::json::string(_part)) +
+
+				_queryr += (!_first ? ", " : "") +
+					   zpt::r_replace(zpt::pgsql::escape(zpt::json::string(_part)), "'", "\"") +
```
Comment (naazgull) [inline postgresql/src/postgresql/convert_sql.cpp]: Could we do this in a more general manner?
Change `zpt::pgsql::escape` to take a second argument, an `std::string`, with default value `'`. The method signature would become:
```c++
auto escape(zpt::json _in, std::string _str_delimiter = "'") -> std::string;
```
and the function body would become:
```c++
auto zpt::pgsql::escape(zpt::json _in, std::string _str_delimiter) -> std::string {
	std::string _out;

	switch (_in->type()) {
	case zpt::JSObject: {
		_out.assign(zpt::pgsql::escape(zpt::json::string(std::string(_in))));
		break;
	}
	case zpt::JSArray: {
		if (_in->arr()->size() == 2 && _in[0]->is_number() && _in[1]->is_number()) {
			_out.assign(std::string("point") +
				    zpt::r_replace(zpt::r_replace(std::string(_in), "[", "("), "]", ")"));
		} else {
			_out.assign(zpt::pgsql::escape(zpt::json::string(std::string(_in))));
		}
		break;
	}
	case zpt::JSString: {
		_out.assign(std::string(_in));
		zpt::replace(_out, _str_delimiter, str_delimiter + str_delimiter);
		_out.insert(0, str_delimiter);
		_out.push_back(str_delimiter{0});
		break;
	}
	case zpt::JSBoolean: {
		_out.assign(std::string(_in));
		break;
	}
	case zpt::JSDouble: {
		_out.assign(std::string(_in));
		break;
	}
	case zpt::JSInteger: {
		_out.assign(std::string(_in));
		break;
	}
	case zpt::JSNil: {
		_out.assign("NULL");
		break;
	}
	case zpt::JSDate: {
		_out.assign(std::string("TIMESTAMP ") + _str_delimiter + std::string(_in) + _str_delimiter);
		break;
	}
	case zpt::JSLambda: {
		_out.assign(std::string(_in));
		break;
	}
	default: {}
	}

	return _out;
}
```

**Comment by vorjdux** [comment]
@naazgull done. I change the suggested code a little but I think that it is much better now :) Thank you for the suggestions!

**Pull request merged**